package/libs/openssl/Makefile

```diff
@@ -95,9 +94,10 @@ $(call Package/openssl/Default)
   DEPENDS:=+OPENSSL_WITH_COMPRESSION:zlib \
 	   +OPENSSL_ENGINE_BUILTIN_AFALG:kmod-crypto-user \
 	   +OPENSSL_ENGINE_BUILTIN_DEVCRYPTO:kmod-cryptodev \
-	   +OPENSSL_ENGINE_BUILTIN_PADLOCK:kmod-crypto-hw-padlock
+	   +OPENSSL_ENGINE_BUILTIN_PADLOCK:kmod-crypto-hw-padlock \
+	   +(arm||armeb||mips||mipsel||ppc):libatomic
```

Comment:
IIRC packaged elf files are checked for NEEDED (as per readelf) for missing dependencies.
Does the reverse work? I.e. always add libatomic here, use LDFLAGS+= -Wl,--as-needed and ipkg only adds the dependency if there's a NEEDED entry?

Reply:
Do you mean, just add an unconditional `+libatomic` dependency?
I don't think ipk tries to sort which dependencies are actually needed based on NEEDED entries.
I tried it, and the ipk will have libatomic1 added to Depends: in package for aarch64, even though OpenSSL does not link to libatomic for 64-bit CPUs.
OpenSSL build configuration files decide which targets use libatomic, so `--as-needed` will not make a difference here.

Reply:
Aww, yeah, I meant that. Bummer, that would've been nicer to maintain.
Thanks for trying it out